pkg/check/permissions.go

```diff
@@ -0,0 +1,326 @@
+	if _, ok := ctx.VMConfig.VirtualCenter[ctx.VMConfig.Workspace.VCenterIP]; !ok {
+		return errors.New("vcenter instance not found in the virtual center map")
+	}
+	username := ctx.VMConfig.VirtualCenter[ctx.VMConfig.Workspace.VCenterIP].User
+
+	matchingDC, err := getDatacenter(ctx, dataCenterName)
+	if err != nil {
+		klog.Errorf("error getting datacenter %s: %v", ctx.VMConfig.Workspace.Datacenter, err)
+		return err
+	}
+	if err := comparePrivileges(ctx.Context, username, matchingDC.Reference(), ctx.AuthManager, permissions[permissionDatacenter]); err != nil {
+		return fmt.Errorf("missing privileges for datacenter %s: %s", dataCenterName, err.Error())
+	}
+	return nil
+}
+
+func checkFolderPrivileges(ctx *CheckContext, folderPath string, group permissionGroup) error {
+	if _, ok := ctx.VMConfig.VirtualCenter[ctx.VMConfig.Workspace.VCenterIP]; !ok {
+		return errors.New("vcenter instance not found in the virtual center map")
+	}
+	username := ctx.VMConfig.VirtualCenter[ctx.VMConfig.Workspace.VCenterIP].User
+
+	finder := find.NewFinder(ctx.VMClient)
+	folder, err := getFolderReference(ctx.Context, folderPath, finder)
+	if err != nil {
+		klog.Errorf("error getting folder %s: %v", folderPath, err)
+		return err
+	}
+	if err := comparePrivileges(ctx.Context, username, folder.Reference(), ctx.AuthManager, permissions[group]); err != nil {
+		return fmt.Errorf("missing privileges for %s: %s", group, err.Error())
+	}
+	return nil
+}
+
+func checkDatastorePrivileges(ctx *CheckContext, dataStoreName string) error {
+	if _, ok := ctx.VMConfig.VirtualCenter[ctx.VMConfig.Workspace.VCenterIP]; !ok {
+		return errors.New("vcenter instance not found in the virtual center map")
+	}
+
+	username := ctx.VMConfig.VirtualCenter[ctx.VMConfig.Workspace.VCenterIP].User
+
+	dc, err := getDatacenter(ctx, ctx.VMConfig.Workspace.Datacenter)
+	if err != nil {
+		klog.Errorf("error getting datacenter %s: %v", ctx.VMConfig.Workspace.Datacenter, err)
+		return err
+	}
+	ds, err := getDataStoreByName(ctx, dataStoreName, dc)
+	if err != nil {
+		klog.Errorf("error getting datastore %s: %v", dataStoreName, err)
+		return err
+	}
+	if err := comparePrivileges(ctx.Context, username, ds.Reference(), ctx.AuthManager, permissions[permissionDatastore]); err != nil {
+		return fmt.Errorf("missing privileges for datastore %s: %s", dataStoreName, err.Error())
+	}
+	return nil
+}
+
+func getFolderReference(ctx context.Context, path string, finder *find.Finder) (*types.ManagedObjectReference, error) {
+	folderObj, err := finder.Folder(ctx, path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "specified folder not found")
+	}
+	ref := folderObj.Reference()
+	return &ref, nil
+}
+
+// CheckAccountPermissions will attempt to validate that the necessary credentials are held by the account performing the
+// installation. each group of privileges will be checked for missing privileges.
+func CheckAccountPermissions(ctx *CheckContext) error {
```

Comment:
nit: Can we move this to a different file? It is bit confusing to have both cluster checks and node checks in same file.

Reply:
We can still keep shared functions in one file or the other. I just feel that, may be this function should be moved to a different file.

Reply:
I'll remove the NodeCheck and we can keep this where it is.